internal/venice-client-common/src/main/java/com/linkedin/venice/schema/SchemaEntry.java

```diff
@@ -144,38 +150,78 @@ public byte[] getSchemaBytes() {
+                this.schema);
+        if (backwardCompatibilityResult.getType() == SchemaCompatibility19.SchemaCompatibilityType.INCOMPATIBLE) {
+          compatible = false;
+          compatibilityResultDescription = backwardCompatibilityResult.getDescription();
+        }
+      } else {
+        SchemaCompatibility.SchemaPairCompatibility backwardCompatibility =
+            SchemaCompatibility.checkReaderWriterCompatibility(
+                /** reader */
+                newSchemaEntry.schema,
+                /** writer */
+                this.schema);
+        if (backwardCompatibility.getType() == INCOMPATIBLE) {
+          compatible = false;
+          compatibilityResultDescription = backwardCompatibility.getDescription();
+        }
+      }
+      if (!compatible) {
         LOGGER.info(
             "New schema (id {}) is not backward compatible with (i.e.: cannot read data written by)"
                 + " existing schema (id {}), Full message:\n{}",
             newSchemaEntry.getId(),
             this.id,
-            backwardCompatibility.getDescription());
+            compatibilityResultDescription);
         return false;
       }
     }
 
     if (Arrays.asList(FORWARD, FULL).contains(expectedCompatibilityType)) {
-      SchemaCompatibility.SchemaPairCompatibility forwardCompatibility = checkReaderWriterCompatibility(
-          /** reader */
-          this.schema,
-          /** writer */
-          newSchemaEntry.schema);
-      if (forwardCompatibility.getType() == INCOMPATIBLE) {
+      if (enumSchemaEvolutionAllowed) {
+        SchemaCompatibility19.SchemaPairCompatibility forwardCompatibilityResult =
+            SchemaCompatibility19.checkReaderWriterCompatibility(
+                /** reader */
+                this.schema,
+                /** writer */
+                newSchemaEntry.schema);
+        if (forwardCompatibilityResult.getType() == SchemaCompatibility19.SchemaCompatibilityType.INCOMPATIBLE) {
+          compatible = false;
+          compatibilityResultDescription = forwardCompatibilityResult.getDescription();
+        }
+      } else {
+        SchemaCompatibility.SchemaPairCompatibility forwardCompatibility =
+            SchemaCompatibility.checkReaderWriterCompatibility(
+                /** reader */
+                this.schema,
+                /** writer */
+                newSchemaEntry.schema);
+        if (forwardCompatibility.getType() == INCOMPATIBLE) {
+          compatible = false;
+          compatibilityResultDescription = forwardCompatibility.getDescription();
+        }
+      }
+      if (!compatible) {
         LOGGER.info(
             "New schema id ({}) is not forward compatible with (i.e.: cannot have its written data read by)"
                 + " existing schema id ({}), Full message:\n{}",
             newSchemaEntry.getId(),
             this.id,
-            forwardCompatibility.getDescription());
+            compatibilityResultDescription);
```

Comment:
nit: can we suggest users to upgrade to 1.9+ if its not obvious from the `compatibilityResultDescription`.

Reply:
Upgrading 1.9 won't solve all the incompatibility failure and enum evolution is only a small fraction.